cmd/main.go

```diff
@@ -407,22 +419,25 @@ func main() {
 		notificationStream, eventsHandler, uploadClient, hostApi, metricsManager, manifestsGenerator, lead, operatorsManager,
 		ocmClient, objectHandler, dnsApi, authHandler, manifestsApi, Options.EnableSoftTimeouts)
 	infraEnvApi := infraenv.NewManager(log.WithField("pkg", "host-state"), db, objectHandler)
+	releaseSourcesAPIHandler := releasesources.NewAPIHandler(releaseSourcesHandler)
 
 	clusterEventsUploader := thread.New(
 		log.WithField("pkg", "cluster-events-uploader"), "Cluster Events Uploader", Options.ClusterEventsUploaderInterval, clusterApi.UploadEvents)
-	clusterEventsUploader.Start()
+	clusterEventsUploader.Start(false)
 	defer clusterEventsUploader.Stop()
 
 	clusterStateMonitor := thread.New(
 		log.WithField("pkg", "cluster-monitor"), "Cluster State Monitor", Options.ClusterStateMonitorInterval, clusterApi.ClusterMonitoring)
-	clusterStateMonitor.Start()
+	clusterStateMonitor.Start(false)
 	defer clusterStateMonitor.Stop()
 
 	hostStateMonitor := thread.New(
 		log.WithField("pkg", "host-monitor"), "Host State Monitor", Options.HostStateMonitorInterval, hostApi.HostMonitoring)
-	hostStateMonitor.Start()
+	hostStateMonitor.Start(false)
 	defer hostStateMonitor.Stop()
 
+	runOpenshiftReleaseSyncerIfNeeded(Options.EnableKubeAPI, Options.EnableOpenshiftReleaseSyncer, releaseSourcesHandler, log)
```

Comment:
What is the intention here? Run once or run now and each interval?
Asking because AFAIU this is running always at most once

Reply:
The intention is run now and each interval, I just extracted it to a function because main was getting "cyclomatic complexity" rate too high

Reply:
I think it's good to extract it to a funciton, the thing is that AFAIU this function closes the thread on completion, so it will run at most once. We should probably defer close from main instead?

Reply:
Oh right sorry, Thank you